test/test-prep.sh

```diff
@@ -29,6 +29,3 @@ remote_code_archive_uri="${code_archive_prefix}_${PULL_BASE_SHA}_${date_string}.
 
 tar -czf "$local_code_archive_file" .
 gsutil cp "$local_code_archive_file" "$remote_code_archive_uri"
-
```

Comment:
Any reason why we move these two variables out of the test-prep script

Reply:
i was having a hard time trace those environment variable as they are spread across multiple files. It's better to locate them in the same top level script.

Reply:
In fact, that is why I tried to move all shared env variable in the test_prep script to reduce code duplication.

Reply:
Shouldn't the environment variable be determined by the caller?  how is it different from other env variables in the top level script?

Reply:
For env variables, yes. I mean, local variables such as PROJECT and GCR_IMAGE_BASE_DIR are not from the caller. TEST_CLUSTER_PREFIX and TEST_CLUSTER are also local variables that are used by these test scripts. 

Above all, there are env variables that are from the caller and there are local variables that presubmit and postsubmit creates. My point was to move these shared local variables in the same script for better reference.

Reply:
I moved them to deploy-kubeflow.sh since it's only used there. 

The variable passing in these scripts start to become a bit casual. Given the complexity of the test script I think we might also consider switching to go/python based binary.

Reply:
SG. Let's create an issue to keep track: https://github.com/kubeflow/pipelines/issues/875